### ISSUE TITLE
"no-unsafe-callback-scope" false positives

### ISSUE THREAD

**Issue by dmytro-gokun** [cartant/tslint-etc#24]
In the latest release (1.5.4), code like this produces warnings:

```
platformBrowserDynamic()
    .bootstrapModule(AppModule)
    .catch(err => console.log(`ERROR CAUGHT: ${err}`));  <== HERE
```

```
platformBrowserDynamic()
    .bootstrapModule(AppModule)
    .catch(() => localStorage.setItem('TOKEN_NAME', 'TOKEN_VALUE'));
```

So, it looks like some global objects, like console, localStorage, sessionStorage need to be whitelisted.

**Comment by cartant** [comment]
TBH, I would consider `localStorage` and `sessionStorage` to be unsafe, as they are side effects.

I think this rule would need to be constrained to callbacks on specific functions to be useful - similar to the way the related RxJS rule is constrained to operators. E.g. a developer might want to have it applied to `Array.prototype.map` or `Array.prototype.filter` only.

ATM, I would not recommend using this rule.

---

For my reference: https://developer.mozilla.org/en-US/docs/Web/API/Window

**Comment by cartant** [comment]
Closed by a4a766a5ae0586f22c6df173af7e2844aa375f0f.
